identify: specify max message size (#492)
libp2p/specs: identify/README.md

| Lifecycle Stage | Maturity Level | Status | Latest Revision |
|-----------------|----------------|--------|-----------------|

> | 3A              | Recommendation | Active | r1, 2022-12-15  |

Reviewer: ```suggestion
| 3A              | Recommendation | Active | r2, 2022-12-15  |
```

}
```

> Implementations must reject `Identify` messages bigger than 4096 bytes.

Reviewer 2: ```suggestion
Implementations SHOULD NOT send `Identify` messages bigger than 4096 bytes.
```

Author: sgtm, will apply
```suggestion
Implementations SHOULD NOT send `Identify` messages bigger than 4096 bytes.
This limit allows receivers to drop messages exceeding the limit, thus better defend against DOS attacks.
```
